src/ManifoldsBase.jl

```diff
@@ -440,34 +440,110 @@ Compute the inner product of tangent vectors `X` and `Y` at point `p` from the
+  Note that this turns `isapprox` basically to an `@assert`.
+* `:info` – prints the information in an `@info`
+* `:warning` – prints the information in an `@warn`
+* `:none` (default) – the function just returns `true`/`false`
+
 Keyword arguments can be used to specify tolerances.
 """
-isapprox(::AbstractManifold, x, y; kwargs...) = isapprox(x, y; kwargs...)
+function isapprox(M::AbstractManifold, p, q; error::Symbol = :none, kwargs...)
+    ma = check_approx(M, p, q; kwargs...)
+    if ma !== nothing
+        (error === :error) && throw(ma)
+        if isnan(ma.val)
+            s = "$(typeof(ma))\n$(ma.msg)"
+        else
+            s = "$(typeof(ma)) with $(ma.val)\n$(ma.msg)"
+        end
+        (error === :info) && @info s
+        (error === :warning) && @warn s
+        return false
+    end
+    return true
+end
+
+function check_approx(M::AbstractManifold, p, q; kwargs...)
+    # fall back to classical approx mode - just that we do not have a reason then
+    res = isapprox(p, q; kwargs...)
+    # since we can not assume distance to be implemented, we can not provide a default value
+    res && return nothing
+    s = "The two points $p and $q on $M are not (approximately) equal."
+    return ApproximatelyError(s)
+end
+
 
 """
-    isapprox(M::AbstractManifold, p, X, Y; kwargs...)
+    isapprox(M::AbstractManifold, p, X, Y; error:Symbol=:none; kwargs...)
 
 Check if vectors `X` and `Y` tangent at `p` from [`AbstractManifold`](@ref) `M` are approximately
 equal.
 
+The optional positional argument can be used to get more information for the case that
+the result is false, if the concrete manifold provides such information.
+Currently the following are supported
+
+* `:error` - throws an error if `isapprox` evaluates to false, providing possibly a more detailed error.
+  Note that this turns `isapprox` basically to an `@assert`.
+* `:info` – prints the information in an `@info`
+* `:warning` – prints the information in an `@warn`
+* `:none` (default) – the function just returns `true`/`false`
+
+By default these informations are collected by calling [`check_approx`](@ref).
+
 Keyword arguments can be used to specify tolerances.
 """
-isapprox(::AbstractManifold, p, X, Y; kwargs...) = isapprox(X, Y; kwargs...)
+function isapprox(M::AbstractManifold, p, X, Y; error::Symbol = :none, kwargs...)
+    mat = check_approx(M, p, X, Y; kwargs...)
+    if mat !== nothing
+        (error === :error) && throw(mat)
+        if isnan(mat.val)
+            s = "$(typeof(mat))\n$(mat.msg)"
+        else
+            s = "$(typeof(mat)) with $(mat.val)\n$(mat.msg)"
+        end
+        (error === :info) && @info s
+        (error === :warning) && @warn s
+        return false
+    end
+    return true
+end
+
+function check_approx(M, p, X, Y; kwargs...)
+    # fall back to classical mode - just that we do not have a reason then
+    res = isapprox(X, Y; kwargs...)
+    res && return nothing
+    v = norm(X - Y)
```

Comment:
```suggestion
    v = norm(M, p, X - Y)
```
I think that's more correct in general.

Reply:
nice idea.

Reply:
The problem with this one is that on the manifold where we can properly test this fallback, said norm is not defined

Reply:
What is the problem with defining that norm?

Reply:
The manifold is called `NonManifold` and we test all MethodErrors with that.

Reply:
But then we can just use a different manifold to test `check_approx`?

Reply:
Sure, we can implement a whole new Manifold with that. All just a question of effort (compared to deleting one line).

Reply:
I can cover that with a test then.

Reply:
Let me just first finish to do the remaining coverage – maybe I just saw one where that can easily be added as well (in domain error we might have a fitting manifold).

My conceptional problem is, that I would like to provide errors without much stuff present, here we would say: For errors to work, please implement a norm, otherwise we error while we error.

Reply:
Hm, we can always wrap norm computation in try/catch then.

Reply:
...or keep it simple. I do not have a strong preference here, but I would either use `norm(X-Y)` or not provide a number (since there is not good default to do so).

Reply:
To me the only good default is `norm(M, p, X - Y)` (if it's implemented) so I prefer no number over `norm(X-Y)` (which may be sometimes different from `norm(M, p, X - Y)`).

Reply:
You are right the actual norm might be misleading especially on quotient.
So we either
a) provide no number
or
b) try/catch norm and provide a number if that suceeds, otherwise no number.

I tend little bit to a) for simplicity, but b) might be nicer in information sometimes (and when we error, performance should not be that much of a problem). You tend to b)?

Reply:
Yes, I prefer b) because it gives additional information which should always be correct but I'm also fine with a). Performance indeed should not be a problem here, this method should not be used in performance-critical code.

Reply:
Ok, I implemented b).

Reply:
Thanks!